ClientRequirments.md

```diff
@@ -0,0 +1,39 @@
+# Platform Requirements for Fluid Framework Client Packages
+
+These are the platform requirements for the current version of Fluid Framework Client Packages.
+These requirements error on the side of being too strict since within a major version they can be relaxed over time, but not made stricter.
+For Long Term Support (LTS) versions this can require supporting these platforms for several years.
```

Comment:
```suggestion
For Long Term Support (LTS) versions this can require meeting these requirements for several years.
```

Reply:
I think that's wrong. Fluid Framework 2.0 would not support the **requirement** that customers use TypeScript 5.4. Instead it supports customers who use TypeScript 5.4 with NodeJS 20 via esm (the platform is what we support, not a requirement: the requirements define a set of platforms which are supported).

Another way to look at this:
Removing a requirement increases what we support, so we are allowed to relax/remove requirements, and thus we aren't required to keep all our requirements over the LTS timeline, so saying "requirement" here seems backwards.

Reply:
I'm open to suggestions on how to increase clarity around this, since I'm sure there is a better way to express this but I haven't found one.

Reply:
Maybe "supported platforms" instead of "platform requirements"?

Reply:
Look at it now, I have no idea how I came to make this suggestion. 🙃
Feel free to resolve without further action.